contrib/clojure-package/src/org/apache/clojure_mxnet/ndarray.clj

```diff
@@ -167,3 +171,36 @@
 
 (defn shape-vec [ndarray]
   (mx-shape/->vec (shape ndarray)))
+
+(s/fdef vec->nd-vec
+        :args (s/cat :v vector? :shape-vec vector?)
+        :ret vector?)
+
+(defn- vec->nd-vec
+  "Convert a vector `v` into a n-dimensional vector given the `shape-vec`
+
+   Ex:
+     (vec->nd-vec [1 2 3] [1 1 3])       ;[[[1 2 3]]]
+     (vec->nd-vec [1 2 3 4 5 6] [2 3 1]) ;[[[1] [2] [3]] [[4] [5] [6]]]
+     (vec->nd-vec [1 2 3 4 5 6] [1 2 3]) ;[[[1 2 3]] [4 5 6]]]
+     (vec->nd-vec [1 2 3 4 5 6] [3 1 2]) ;[[[1 2]] [[3 4]] [[5 6]]]
+     (vec->nd-vec [1 2 3 4 5 6] [3 2])   ;[[1 2] [3 4] [5 6]]"
+  [v [s1 & ss :as shape-vec]]
+  (if-not (seq ss)
+    (vec v)
+    (->> v
+         (partition (/ (count v) s1))
+         vec
+         (mapv #(vec->nd-vec % ss)))))
+
+(s/fdef ->nd-vec
+        :args (s/cat :ndarray #(instance? NDArray %))
+        :ret vector?)
+
+(defn ->nd-vec
+  "Convert an ndarray `nd` into a n-dimensional Clojure vector.
+  Ex:
+    (->nd-vec (array [1] [1 1 1]))             ;[[[1.0]]]
+    (->nd-vec (ndarray/array [1 2 3] [3 1 1])) ;[[[1.0]] [[2.0]] [[3.0]]]"
+  [ndarray]
```

Comment:
I like the function spec, do you think it would also be nice to be able to do a `util/validate` in here to check that it is a NDArray? when called without instrumentation?

Reply:
Yes lets do that!